CONTRIBUTING.md

```diff
@@ -1,69 +1,130 @@
+
+Try to do one pull request per change. The time taken to review a PR grows exponential with the size
+of the change. Small focused PRs will generally be much more faster to approved and merge. Don't
+include large refactoring or formatting changes in PRs
+
+### Search tui-rs for similar work
 
-### Closing Issues
+Tui-rs has a large amount of history in issues, discussions and PRs that is relevant to this fork.
+Please make sure to search and link any previous issues / discussions related to your PR.
 
-If you close an issue that you have "imported" to this fork, please make sure that you add the issue to the **CLOSED_ISSUES.md**. This will enable us to keep track of which issues have been closed from the original repo, in case we are able to have the original repository transferred.
+### Use conventional commits
 
-# Contributing
+We use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) and check for them as
+a lint build step. To help adhere to the format, we recommend to install
+[Commitizen](https://commitizen-tools.github.io/commitizen/). By using this tool you automatically
+follow the configuration defined in [.cz.toml](.cz.toml). Your commit messages should have enough
+information to help someone reading the [CHANGELOG](./CHANGELOG.md) understand what is new just from
+the title. The summary helps expand on that to provide information that helps provide more context,
+describes the nature of the problem that the commit is solving and any unintuitive effects of the
+change. It's rare that code changes can easily communicate intent, so make sure this is clearly
+documented here.
+
+When the PR is close to approval, please ensure that it is rebased against main, squashed into an
+appropriate amount of commits (usually 1). We don't need to have a commit for every small change
+that was made during the PR process. Use `git push --force` to update the branch associated with
+your PR in your fork.
+
+### Run CI tests before pushing a PR
+
+We're using [cargo-husky](https://github.com/rhysd/cargo-husky) to automatically run git hooks,
+which will run `cargo make ci` before each push. To initialize the hook  run `cargo test`. If
+`cargo-make` is not installed, it will provide instructions to install it for you. This will ensure
+that your code is formatted, compiles and passes all tests before you push. If you need to skip this
+check, you can use `git push --no-verify`.
 
 ## Implementation Guidelines
 
-### Use of unsafe for optimization purposes
+### Setup
 
-**Do not** use unsafe to achieve better performances. This is subject to change, [see.](https://github.com/tui-rs-revival/tui-rs-revival/discussions/66)
-The only exception to this rule is if it's to fix **reproducible slowness.**
+Clone the repo and build it using [cargo-make](https://sagiegurari.github.io/cargo-make/)
 
-## Building
+`ratatui` is an ordinary Rust project where common tasks are managed with [cargo-make]. It wraps
```

Comment:
Oops, can you add a link to the `cargo-make` here?

Reply:
Done